core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java

```diff
@@ -708,7 +708,7 @@ public static boolean isBlockInvalid(SegmentStatus blockStatus) {
    * @return
    */
   public static long readCurrentTime() {
-    return System.currentTimeMillis();
+    return System.nanoTime();
```

Comment:
This will impact the complete system if you change to nano here. try changing the task number to nao instead of here